docs/overrides.rst

```diff
@@ -0,0 +1,130 @@
+Overrides
+=========
+
+During an image life cycle there can be a need to do a slightly tweaked builds - using different base images, injecting newer libraries etc. We want to support such scenarios without a need of duplicating whole image sources. To achieve this Concreate supports overrides mechanism for its image descriptor. You can override almost anything in image descriptor. The overrides are based on overrides descriptor - a yaml file containing changes to image descriptor.
+
+To use an override descriptor you need to pass ``--overides <PATH>`` argument to a concreate.
+
+**Example**: To use overrides.yaml  file located in current working directory run:
+
+.. code:: bash
+
+	  $ concreate build --overrides=overrides.yaml
+
+How overrides works
+-------------------
+
+Concreate is using `YAML <http://yaml.org/>`_ format for its descriptors. Overrides in concreate works on `YAML node <http://www.yaml.org/spec/1.2/spec.html#id2764044>`_ level.
+
+
+Scalar nodes
+^^^^^^^^^^^^
+Scalar nodes are easy to overrides, if Concreate finds any scalar node in a overrides descriptor it update its value in image descriptor with the overridden one.
+
+**Example**: Overriding scalar node:
+
+*image descriptor*
```

Comment:
Maybe instead of this use: `image.yaml` and `override.yaml`? This will play nicely together with the example at the top.

Reply:
If I do this, how will I call the third one? :)

Reply:
`result.yaml`?